src/App.tsx

```diff
@@ -1,24 +1,292 @@
 
-const USER_ID = 0;
+const USER_ID = 10587;
 
 export const App: React.FC = () => {
+  const [errorMessage, setErrorMessage] = useState('');
+  const [filter, setFilter] = useState<FilterType>(FilterType.All);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [newTodo, setNewTodo] = useState('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isUpdating, setIsUpdating] = useState<number[]>([]);
+  const [inputDisabled, setInputDisabled] = useState(false);
+
+  const handleCloseErrorMessage = () => setErrorMessage('');
+
+  const handleLoadTodos = useCallback(async () => {
+    try {
+      const todosFromApi = await getTodos(USER_ID);
+
+      setTodos(todosFromApi);
+    } catch {
+      setErrorMessage(ErrorType.Load);
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    }
+  }, []);
+
+  useEffect(() => {
+    handleLoadTodos();
+  }, []);
+
+  const completedTodos = useMemo(() => todos.filter(todo => todo.completed), [todos]);
+  const activeTodos = useMemo(() => todos.filter(todo => !todo.completed), [todos]);
+
+  const filteredTodos = useMemo(() => {
+    return todos.filter((todo: Todo) => {
+      switch (filter) {
+        case FilterType.Completed:
+          return todo.completed;
+        case FilterType.Active:
+          return !todo.completed;
+        default:
+          return todo;
+      }
+    });
+  }, [todos, filter]);
+
+  const handleNewTodoChange = useCallback((event: React.ChangeEvent<HTMLInputElement>) => {
+    setNewTodo(event.target.value);
+  }, []);
+
+  const addNewTodo = async (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setInputDisabled(true);
+
+    if (newTodo.trim() === '') {
+      setErrorMessage(ErrorType.EmptyTitle);
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+
+      return;
+    }
+
+    try {
+      const userTodo = {
+        id: 0,
+        userId: USER_ID,
+        title: newTodo,
+        completed: false,
+      };
+
+      setIsUpdating(prev => [...prev, userTodo.id]);
+      setTempTodo(userTodo);
+      const addedTodo = await addTodo(userTodo);
+
+      setTodos((prev) => [...prev, addedTodo]);
+    } catch (err) {
+      setErrorMessage(ErrorType.Add);
+      setTimeout(() => {
+        setErrorMessage('');
+      }, 3000);
+    } finally {
+      setTempTodo(null);
+      setNewTodo('');
+      setInputDisabled(false);
+    }
+  };
+
+  const onDeleteTodo = async (id: number) => {
+    setInputDisabled(true);
+    setIsUpdating(prev => [...prev, id]);
+
+    try {
+      await deleteTodo(id);
+      setTodos((prev) => prev.filter((todo) => todo.id !== id));
+    } catch {
+      setErrorMessage(ErrorType.Delete);
+    } finally {
+      setInputDisabled(false);
+      setIsUpdating([]);
+    }
+  };
+
+  const completedTodosId = useMemo(() => completedTodos.map(todo => todo.id), [todos]);
+
+  const onDeleteCompleted = async () => {
```

Comment:
lets name functions according to checklist pt 6 - it should start with `handle` word